eth/downloader/beaconsync.go

```diff
@@ -256,8 +254,6 @@ func (d *Downloader) findBeaconAncestor() (uint64, error) {
 			known = d.blockchain.HasBlock(h.Hash(), n)
 		case SnapSync:
 			known = d.blockchain.HasFastBlock(h.Hash(), n)
-		default:
-			known = d.lightchain.HasHeader(h.Hash(), n)
```

Comment:
Same, no default clause will blow up in very weird ways

Reply:
Note, if this is internal and you already validated that only the first two can happen, then use a panic as the default. Anything really to make it future proof.